compiler/rustc_parse/src/parser/item.rs

```diff
@@ -1109,6 +1113,75 @@ impl<'a> Parser<'a> {
 
         let enum_definition =
             EnumDef { variants: variants.into_iter().filter_map(|v| v).collect() };
+
+        // Implicitly add a #[derive(core::enums::AsRepr)] attribute to all data-free enums with an
+        // explicit repr.
+        //
+        // This is currently cfg!(not(bootstrap)) gated because `core::enums::AsRepr` does not exist on bootstrap,
```

Comment:
I realise using `cfg!(not(bootstrap))` in rustc is weird and slightly frowned upon - if we wanted to avoid this, we could leave the top commit (doing the auto-deriving) out of this PR, and re-add it as a separate PR once the rest of this one hits beta.

Reply:
It is the standard library that needs to be compatible with both the bootstrap and in-tree compiler during bootstrapping. The in-tree compiler will only compile anything against the in-tree standard library, which does contain `core::enums::AsRepr`.

Reply:
Right you are - fixed!